File: src/main/java/de/churl/cookbook/utility/IdUtil.java
package de.churl.cookbook.utility;

import java.util.Collection;
import java.util.UUID;
import java.util.stream.Collectors;

public final class IdUtil {

    private IdUtil() {}

    public static Collection<UUID> toUUID(Collection<String> ids) {
        return ids.stream()
                  .map(UUID::fromString)
                  .collect(Collectors.toUnmodifiableSet());
    }

    public static UUID toUUID(String id) {
        return UUID.fromString(id);
    }

    public static Collection<String> toString(Collection<UUID> ids) {
        return ids.stream()
                  .map(UUID::toString)
                  .collect(Collectors.toUnmodifiableSet());
    }

    public static String toString(UUID id) {
        return id.toString();
    }
}

File: src/main/kotlin/de/churl/cookbook/util/IdUtil.kt
package de.churl.cookbook.util

import java.util.*

fun toUUID(ids: Collection<String>): Collection<UUID> {
    return ids.map { UUID.fromString(it) }
}

fun toUUID(id: String): UUID {
    return UUID.fromString(id)
}

fun emptyUUID(): UUID {
    return UUID.fromString("00000000-0000-0000-0000-000000000000")
}

fun UUIDisEmpty(id: UUID): Boolean {
    return id.equals(emptyUUID())
}
